pf/tfgen/main.go

```diff
@@ -60,6 +70,171 @@ func Main(provider string, info tfbridge.ProviderInfo) {
 	})
 }
 
+// Implements main() logic for a multi-provider build-time helper utility. By convention these utilities are
+// named pulumi-tfgen-$provider, for example when building a "random" provider the program would be called
+// pulumi-tfgen-random.
+//
+// The resulting binary is able to generate [Pulumi Package Schema] as well as provider SDK sources in various
+// programming languages supported by Pulumi such as TypeScript, Go, and Python.
+//
+// [Pulumi Package Schema]: https://www.pulumi.com/docs/guides/pulumi-packages/schema/
+func MainWithMuxer(provider string, infos ...tfbridge.Muxed) {
```

Comment:
Mark this experimental please in the comment. Something in the API/package breakdown is a bit fishy, but not worth blocking the PR on.